kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/system/SuperSystemService.java

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Eurotech and/or its affiliates and others
+ *
+ *   All rights reserved. This program and the accompanying materials
+ *   are made available under the terms of the Eclipse Public License v1.0
+ *   which accompanies this distribution, and is available at
+ *   http://www.eclipse.org/legal/epl-v10.html
+ ******************************************************************************/
+package org.eclipse.kura.core.system;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.eclipse.kura.core.util.ProcessUtil;
+import org.eclipse.kura.core.util.SafeProcess;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Superclass of System*Service.
+ */
+public class SuperSystemService {
+
+    private static final Logger logger = LoggerFactory.getLogger(SuperSystemService.class);
+
+    protected String runSystemCommand(String command) {
+        return runSystemCommand(command.split("\\s+"));
+    }
+
+    protected static String runSystemCommand(String[] commands) {
```

Comment:
Instead of making it a statefull class, you can make it an utility class containing `runSystemCommand(String[])` and `runSystemCommand(String)` static methods. In such scenario, you don't have to extend this class, rather you can use the methods statically in `SystemAdminServiceImpl` and `SystemServiceImpl`.